### ISSUE TITLE
ODS in a Box

### ISSUE THREAD

**Issue by felipecruz91** [opendevstack/ods-core#936]
The concept of ODS in a box it's a really great idea and I am learning more about it in my spare time. I am interested in creating a cloud-agnostic VM image (not coupled to AWS for instance) that can run on my laptop (not in the cloud) using [Vagrant](https://www.vagrantup.com/) with VirtualBox.

I've forked `ods-core` (master branch) into my personal account, made the pertinent changes, and run `packer build -on-error=ask ./ods-devenv/packer/CentOS2ODSBoxVagrant.json`

It installs the Atlassian stack, OpenShift, etc but it fails when it tries to build the nexus image:

**Screenshots**

![nexus-build-error](https://user-images.githubusercontent.com/15997951/103633129-99680180-4f45-11eb-9cc6-78a9971085a0.PNG)

![nexus-build-events](https://user-images.githubusercontent.com/15997951/103633310-da601600-4f45-11eb-8514-de9dec66aa02.PNG)

**Affected version (please complete the following information):**
 - OpenShift: 3.11
 - OpenDevStack (master branch)

**Additional context**
From the error message, I understand that the user does not have the RBAC permissions needed to list the secrets under the ods namespace.

**Comment by felipecruz91** [comment]
After this failure, I have triggered the nexus build with `oc start-build nexus -n ods` and now it has completed successfully. I _guess_ it is a timing issue where the secrets are not created by the time the build starts.

Perhaps it would be interesting to wait for all the resources to be created in OpenShift before starting the [setup_nexus](https://github.com/felipecruz91/ods-core/blob/feature/ods-box/ods-devenv/scripts/deploy.sh#L1887) function.

**Comment by clemensutschig** [comment]
Do you have the err message from the failed run?

Felipe Cruz Martinez <notifications@github.com> schrieb am Di., 5. Jän.
2021, 11:48:

**Comment by clemensutschig** [comment]
Can you try this again with a wait/sleep before the startbuild?

Felipe Cruz Martinez <notifications@github.com> schrieb am Di., 5. Jän.
2021, 11:22:

**Comment by felipecruz91** [comment]
@clemensutschig It seems to be a timing issue, yes.

In this second run, it has progressed a lot and has started testing the ODS installation but it seems that the Prov App is not running:

```cli
    vagrant: === RUN   TestVerifyOdsProjectProvisionThruProvisionApi
    vagrant:     provision-api_test.go:40: Failed to delete project: Execution of `provisioning-app-api.sh` for 'DELETE/ODSVERIFY' failed:
    vagrant:         StdOut: No config file found, assuming defaults, current dir: /home/openshift/opendevstack/ods-core/tests/smoketest
    vagrant:
    vagrant:         Started provision project/component script with command (DELETE)!
    vagrant:
    vagrant:         ... encoding basic auth credentials in base64 format
    vagrant:
    vagrant:         ... sending request to 'https://prov-app-ods.ocp.odsbox.lan' (output will be saved in file './response.txt' and headers in file './headers.txt')


(omitted logs)


    vagrant:         < HTTP/1.0 503 Service Unavailable
    vagrant:         < Pragma: no-cache
    vagrant:         < Cache-Control: private, max-age=0, no-cache, no-store
    vagrant:         < Connection: close
    vagrant:         < Content-Type: text/html
    vagrant:         <
    vagrant:         { [data not shown]
    vagrant: 100  3265    0  3265    0     0   7805      0 --:--:-- --:--:-- --:--:--  7811
    vagrant:         * Closing connection 0
    vagrant:         Error from server (NotFound): namespaces "ODSVERIFY" not found
    vagrant:
    vagrant:         Err: exit status 1
    vagrant: --- FAIL: TestVerifyOdsProjectProvisionThruProvisionApi (2.52s)
```

After some [investigation](https://gist.github.com/felipecruz91/1c124fe2fb24ec3ab341dc93b8256751#file-ods-box-vagrant-logs-txt-L36-L47), it seems that the DeploymentConfig resource of the Prov App is not created by Tailor.

**Comment by michaelsauter** [comment]
@felipecruz91 Could be a resource issue ... https://gist.github.com/felipecruz91/1c124fe2fb24ec3ab341dc93b8256751#file-ods-box-vagrant-logs-txt-L126-L148 shows that the build is `pending` and never progresses ...

**Comment by felipecruz91** [comment]
@michaelsauter Yes, the lack of resources makes the build take longer than expected 😅 but it ends up completing successfully after ~ 5 min. The problem is that the script bails out _before_ completing and the rest of resources such as the DeploymentConfig are not applied  (just right after `make apply-provisioning-app-deploy`)

**Comment by clemensutschig** [comment]
All of the startbuild/rollout habe --follow.. you could try to set/increase
the timeout

Felipe Cruz Martinez <notifications@github.com> schrieb am Di., 5. Jän.
2021, 17:00:

**Comment by felipecruz91** [comment]
Increasing the timeout solves this particular issue. As a suggestion for the future, it would be nice to monitor the build status and make the script _wait_ until the build has successfully completed (instead of waiting a fixed time and hope the build will complete within the timeout period).

**Comment by clemensutschig** [comment]
Felipe - feel tree to Change an Open a PR ;-)

Felipe Cruz Martinez <notifications@github.com> schrieb am Do., 7. Jän.
2021, 11:18:
